pkg/liveshare/port_forwarder_test.go

```diff
@@ -85,7 +85,9 @@ func TestPortForwarderStart(t *testing.T) {
 		retries := 0
 		for conn == nil && retries < 2 {
 			conn, err = net.DialTimeout("tcp", ":8000", 2*time.Second)
-			time.Sleep(1 * time.Second)
+			if conn == nil {
+				time.Sleep(1 * time.Second)
+			}
```

Comment:
Shouldn't this loop increment retries?

Reply:
Also: why is the loop necessary? Under what circumstances does retrying DialTimeout succeed in a loop but fail if called once with a larger timeout? That seems like a bug. The only things that should cause DialTimeout to fail fast are intrinsic failures (e.g. bad arguments), not things that depend on the state of the network.

Reply:
Gosh, this was a while ago so I can't remember the specifics, but I remember us discussing that there is an inherent race from the time that the port forwarder is started to the time it is available for connections. I think we had decided to implement this retry strategy to deal with this.

The retry increment is indeed missing.